core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/provision/service/ClusterProxyService.java

```diff
@@ -225,8 +243,16 @@ private String internalAdminUrl(Stack stack, int port) {
         return String.format("https://%s:%d", gatewayIp, port);
     }
 
-    public String getProxyPath(String crn) {
-        return String.format("%s/proxy/%s/%s", clusterProxyConfiguration.getClusterProxyBasePath(), crn, CB_INTERNAL);
+    public String getProxyPath(String crn, String internalIdentity) {
```

Comment:
could you please add logging? also I would move out the decesion logic to a separate method:
```
    public String getProxyPath(String crn, String internalIdentity) {
        String endpointWithIdentity = CB_INTERNAL + "-" + internalIdentity;
        if (isServiceEndpointWithIdentityRegistered(crn, endpointWithIdentity)) {
            <LOG>
            return String.format("%s/proxy/%s/%s", clusterProxyConfiguration.getClusterProxyBasePath(), crn, endpointWithIdentity);
        } else {
            <LOG>
            return String.format("%s/proxy/%s/%s", clusterProxyConfiguration.getClusterProxyBasePath(), crn, CB_INTERNAL);
        }
    }

    private boolean isServiceEndpointWithIdentityRegistered(String crn, String endpointWithIdentity) {
        ReadConfigResponse readConfigResponse = clusterProxyRegistrationClient.readConfig(crn);
        return readConfigResponse.getServices().stream()
                .anyMatch(readConfigService -> endpointWithIdentity.equals(readConfigService.getName()));
    }